### DISCUSSION TITLE
How to enable key-repeat for Vim extension?

### DISCUSSION THREAD

**Discussion by jsjoeio** [coder/code-server#2271] [Q&A] [answered]
Normally, I use the [Vim extension](https://github.com/VSCodeVim/Vim) for VSCode and enable key-repeating with:

```shell
defaults write com.microsoft.VSCode ApplePressAndHoldEnabled -bool false
```

Obviously, that wouldn't work with `code-server` since it's running as a PWA via Chrome.

How do I enable key-repeating so that it works within `code-server`?

Much thanks in advance!

**Comment by code-asher** [comment]
I don't have a Mac so my knowledge here is severely limited but I think
the only way might be to disable it for Chrome or globally. There
doesn't seem to be an option to disable it for a specific website or
target a PWA or anything like that.

**Reply by jsjoeio** [reply]
Firefox is my main browser anyway so that might be the easiest bet. I'll test that out and report back.

**Comment by jsjoeio** [comment, marked as answer]
Disabling it globally worked (as suggested by @code-asher) 

On macOS, you can do this by running:
```shell
defaults write -g ApplePressAndHoldEnabled -bool false
```

[Source](https://support.rstudio.com/hc/en-us/articles/204896737-Fix-Vim-Mode-Repeating-Keys-on-Mac-OS-X)
